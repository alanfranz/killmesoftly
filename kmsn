#!/usr/bin/env bash
. "$(dirname $0)/kms_functions"

[ -z "$1" ] && { 
    echo -e "Kill processes softly by name matching pattern."
    echo -e "usage: $(basename "${0}") first_pgrep_pattern [second_pgrep_pattern] <...> \n"
    exit 1
}

while (( "$#" )); do
    PROCESS_PIDS=($(pgrep $1))
    [ "${#PROCESS_PIDS[@]}" -gt 1 ] && { echo "More than one process matches $1, won't kill multiple processes"; shift; continue; }
    [ "${#PROCESS_PIDS[@]}" -eq 0 ] && { echo "No process matches given pattern $1"; shift; continue; }
    PID=${PROCESS_PIDS[0]}
    killmesoftly ${PID}
    shift
done
