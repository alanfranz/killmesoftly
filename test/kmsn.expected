No process matches given pattern notexisting
Now sending SIGTERM to PID X... didn't terminate, sending SIGKILL... killed successfully
Now sending SIGTERM to PID X... killed successfully
Now sending SIGTERM to PID X... killed successfully
Now sending SIGTERM to PID X... killed successfully
